src/jit/simd.h

```diff
@@ -5,6 +5,42 @@
 #ifndef _SIMD_H_
 #define _SIMD_H_
 
+// Underlying hardware information
+// This type is used to control
+// 1. The length of System.Numerics.Vector<T>.
+// 2. Codegen of System.Numerics.Vectors.
+// 3. Codgen of floating-point arithmetics (VEX-encoding or not).
+//
+// Note
+// - Hardware SIMD support is classified to the levels. Do not directly use
+//   InstructionSet (instr.h) for System.Numerics.Vectors.
+// - Values of SIMDLevel have strictly increasing order.
+enum SIMDLevel
+{
+    SIMD_Not_Supported = 0,
+#ifdef _TARGET_XARCH_
+    // SSE2 - The min bar of SIMD ISA on x86/x64.
+    // Vector<T> length is 128-bit.
+    // Floating-point instructions are legacy SSE encoded.
+    SIMD_SSE2_Supported = 1,
+
+    // SSE3_4 - RyuJIT generates SSE3, SSSE3, SSE4.1 and SSE4.2 instructions for certain intrinsics.
+    // Vector<T> length is 128-bit.
+    // Floating-point instructions are legacy SSE encoded.
+    SIMD_SSE3_4_Supported = 2,
```

Comment:
Question: would this be better named SIMD_SSE42_Supported? In general, these names seem to indicate a maximum supported level. Is it true that any Intel machine with SSE4.2 also has support for SSE3/SSSE3/SSE4.1 instructions that we generate?

Reply:
We have had the name "SSE3_4" exposed outside (i.e. `EnableSSE3_4`). Shall we keep consistent name here?
Question: can we remove `CORJIT_FLAG_USE_SSE3_4` and the related CPUID checking in codeman.cpp? Then we can set `compCanUseSSE3_4` by `compSupports(InstructionSet_SSE41) && compSupports(InstructionSet_SSE42)`. But I worry about if the change breaks Desktop CLR code base.

Reply:
@CarolEidt any comment?

Reply:
I don't think the names of the CORJIT_FLAG values need to be consistent with the COMPlus_EnableSES3_4. Am I missing something? That is, I think we can change these names internally without breaking the support for the flag.

Reply:
OK, I would like to change the name to ` SIMD_SSE4_Supported` that means having SSE4.1 and SSE4.2 both. For `CORJIT_FLAG_XXX`, can we remove `CORJIT_FLAG_USE_SSE3_4` in the VM and just define it in JIT?

Reply:
> For CORJIT_FLAG_XXX, can we remove CORJIT_FLAG_USE_SSE3_4 in the VM and just define it in JIT?

Don't we still need the VM to tell us if we're on a SSE4.2 capable machine? It uses that flag to do so.

Reply:
Now, we have all the ISA flags (introduced by previous HW intrinsic PR) in the VM side.

Reply:
Then maybe cleaning up the `CORJIT_FLAG_XXX` could be a separate PR?

Reply:
(Note that changing the JIT-EE interface, of which these are a part, has some internal Microsoft overhead for us that is not obvious in GitHub)

Reply:
I will watch for this when it gets checked-in

Reply:
@BruceForstall @briansull Thanks for the information. Will let you know when I make the jit-ee interface change.